bin/newman.js

```diff
@@ -62,11 +62,64 @@ program
     .option('--export-cookie-jar <path>', 'Exports the cookie jar to a file after completing the run')
     .option('--verbose', 'Show detailed information of collection run and each request sent')
     .action((collection, command) => {
+        // Process reporter arguments explicitly
+        const rawArgs = command.parent.args,
+            isReporterPresent = rawArgs.includes('-r') || rawArgs.includes('--reporter');
+
+        if (isReporterPresent) {
+            const reporters = ['cli'];
+            let reporterIndex = rawArgs.indexOf('-r'),
+                reporterIterator;
+
+            if (reporterIndex === -1) {
+                reporterIndex = rawArgs.indexOf('--reporter');
+            }
+
+            reporterIterator = reporterIndex + 1;
+
+            while (reporterIterator < rawArgs.length) {
+                const reporterName = rawArgs[reporterIterator];
+
+                if (reporterName.slice(0, 1) === '-') {
+                    break;
+                }
+
+                if (reporterName.includes(',')) {
+                    reporterName.split(',').forEach((reporterName) => {
+                        if (reporterName.length) {
+                            reporters.push(reporterName);
+                        }
+                    });
+                }
+                else {
+                    reporters.push(reporterName);
+                }
+
+                if (reporterName.slice(-1) !== ',') {
+                    break;
+                }
+
+                reporterIterator += 1;
+            }
+
+            // Trim all the reporter names for white spaces and commas
+            reporters.forEach((reporterName, reporterIndex, reporters) => {
+                let sanitizedReporterName = reporterName.trim();
+
+                if (sanitizedReporterName.slice(-1) === ',') {
+                    sanitizedReporterName = sanitizedReporterName.split(',').join('');
+                }
+
+                reporters[reporterIndex] = sanitizedReporterName;
+            });
+
+            command.reporters = reporters;
+        }
         let options = util.commanderToObject(command),
```

Comment:
What does this util do? Asking since I want to check if the pattern is to put put reusable CLI processing in util. Which is also not a good pattern, but still ... asking.

Reply:
@shamasis This is what the method comments say
```js
/**
     * Extract selected options in the provided command.
     * Omits commander private variables and other objects.
     *
     * @param {Object} command - Commander.Command Instance
     * @returns {Object} - Extracted options from command
     */
```

I think CLI processing is in util only.